notebooks/kaggle/store-sales-time-series-forecasting/experiments/LightGBM.ipynb

```diff
@@ -640,3107 +292,854 @@
+    "    if current_family in linear_categories:\n",
+    "        base_pipelines[current_family] = LinearPipeline(cols_to_scale=['dcoilwtico'], cols_to_encode=['store_nbr'], \n",
+    "                                                         drop_columns=['onpromotion'], lags=lags, split_key='family', \n",
+    "                                                         target_col='sales')\n",
+    "    else:\n",
+    "        base_pipelines[current_family] = LGBMPipeline(drop_columns=drop_columns[current_family], lags=lags, \n",
+    "                                                       split_key='family', target_col='sales', fit_params=fit_params, \n",
+    "                                                       params=best_final_params_boruta[current_family])\n",
+    "        if 'store_nbr' in drop_columns[current_family]:\n",
+    "            base_pipelines[current_family].fit_params = {}\n",
+    "            \n",
+    "final_modelling_pipeline = RecursiveTSEstimator(split_key='family', target_col='sales', base_pipelines=base_pipelines,\n",
+    "                                                 zero_categories=['baby care', 'books'])\n",
+    "\n",
+    "final_modelling_pipeline.fit(X_train, y_train)\n",
+    "make_submission_file(test_data, final_modelling_pipeline, 'optuna_LGBM_and_linreg_boruta.csv')"
    ]
   },
   {
-   "cell_type": "code",
-   "execution_count": null,
-   "id": "0af3cb82",
+   "cell_type": "markdown",
+   "id": "9460c4f3",
    "metadata": {},
-   "outputs": [],
    "source": [
-    "y = X['sales'].copy()\n",
-    "X = X.drop(columns=['sales'])\n",
-    "X['sales'] = y"
+    "### Score on Kaggle: 0.43458"
    ]
   },
   {
-   "cell_type": "code",
-   "execution_count": null,
-   "id": "c6b619ac",
+   "cell_type": "markdown",
+   "id": "5069ea0a",
    "metadata": {},
-   "outputs": [],
    "source": [
-    "hyper_params = {\n",
-    "    'max_depth': 6, \n",
-    "    'n_estimators': 300,\n",
-    "    'num_leaves': 40,\n",
-    "    'reg_alpha': 0.1,\n",
-    "    'reg_lambda': 0.1\n",
-    "}"
+    "### Adding features from other datasets"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "id": "cd5ba7b4",
+   "id": "978f2d5e",
    "metadata": {},
    "outputs": [],
    "source": [
-    "first_day_of_last_year = pd.to_datetime(X['date'].unique()[-1]) - timedelta(days=365)\n",
-    "indexer = X[X['date'] >= str(first_day_of_last_year).split(' ')[0]].index\n",
-    "X_train = X[X['date'] >= str(first_day_of_last_year).split(' ')[0]]\n",
-    "y_train = y.loc[indexer]"
+    "X_copy = X.copy()\n",
+    "# X_copy['is_holiday'] = train_transformer.adding_is_holiday_feature(X_copy)\n",
+    "X_copy = train_transformer.adding_stores_data(X_copy, columns_to_add=['city', 'type'])\n",
+    "X_copy['store_city'] = LabelEncoder().fit_transform(X_copy['store_city']) + 1\n",
+    "X_copy['store_type'] = LabelEncoder().fit_transform(X_copy['store_type']) + 1\n",
```

Comment:
I decided to add 1 to the encoded values just to start encoding from 1, not from 0

Reply:
Why is it important to start encoding from 1?

Reply:
Maybe if we are talking about linear models, it is important to start encoding from 1, because for samples encoded with 0 the corresponding component in the linear combination will be 0 (which can not be trustful in some cases)